unittests/ThunkLibs/common.h

```diff
@@ -42,15 +42,38 @@ class TestDiagnosticConsumer : public clang::TextDiagnosticPrinter {
     }
 };
 
+enum class GuestABI {
+    X86_32,
+    X86_64,
+};
+
+inline std::ostream& operator<<(std::ostream& os, GuestABI abi) {
+    if (abi == GuestABI::X86_32) {
+        os << "X86_32";
+    } else if (abi == GuestABI::X86_64) {
+        os << "X86_64";
+    }
+    return os;
+}
+
 /**
  * Run the given ToolAction on the input code.
  *
  * The "silent" parameter is used to suppress non-fatal diagnostics in tests that expect failure
  */
-inline void run_tool(clang::tooling::ToolAction& action, std::string_view code, bool silent = false) {
+inline void run_tool(clang::tooling::ToolAction& action, std::string_view code, bool silent = false, std::optional<GuestABI> guest_abi = std::nullopt) {
```

Comment:
libclang will leak memory in `invocation.run()` when the given `ToolAction` throws an exception. This is annoying but otherwise harmless. We may consider defining something like this here though to disable memory leak checking for thunk unit tests:
```cpp
const char* __asan_default_options() { return "detect_leaks=0"; }
```